cloudferrylib/os/identity/keystone.py

```diff
@@ -130,13 +131,22 @@ def read_info(self, **kwargs):
         service_tenant_id = \
             self.get_tenant_id_by_name(self.config.cloud.service_tenant)
 
+        admin_tenant_id = \
+            self.get_tenant_id_by_name(self.config.cloud.admin_tenant)
+
+        if kwargs.get('tenant_id'):
+            self.filter_tenant_id = kwargs['tenant_id'][0]
+
+        tmp_list = [admin_tenant_id, service_tenant_id, self.filter_tenant_id]
         for tenant in self.get_tenants_list():
-            if tenant.id != service_tenant_id:
+            if (self.filter_tenant_id and tenant.id in tmp_list) or \
+                    not self.filter_tenant_id:
```

Comment:
Before this, service_tenant_id parameter was used to ignore the tenant service, and now there is.